# Assessment Problems

## Problem 1: Extending the Lady Tasting Tea

### Description of the problem

Let's extend the Lady Tasting Tea experiment as follows. The original experiment has 8 cups: 4 tea-first and 4 milk-first. Suppose we prepare 12 cups: 8 tea-first and 4 milk-first. A participant claims they can tell which was poured first.

Simulate this experiment using numpy by randomly shuffling the cups many times and calculating the probability of the participant correctly identifying all cups by chance. Compare your result with the original 8-cup experiment.

In the notebook, explain your simulation process clearly, report and interpret the estimated probability, and discuss whether, based on this probability, you would consider extending or relaxing the p-value threshold compared to the original design.

### Libraries

In [1]:
# Mathematical functions from the standard library.
# https://docs.python.org/3/library/math.html
import math

# Permutations and combinations.
# https://docs.python.org/3/library/itertools.html
import itertools

# Random selections.
# https://docs.python.org/3/library/random.html
import random

# Numerical structures and operations.
# https://numpy.org/doc/stable/reference/index.html#reference
import numpy as np

# Plotting.
# https://matplotlib.org/stable/contents.html
import matplotlib.pyplot as plt

### Description of the original experiment. 

In the original experiment the lady claims she can know by tasting the tea cups if the milk or the tea was poured first. The experiment is based with 8 cups: 4 tea-first, 4 milk-first).

#### Cups of tea

In [ ]:
# Number of cups of tea in total.
no_cups = 8

# Number of cups of tea with milk in first.
no_cups_milk_first = 4

# Number of cups of tea with tea in first.
no_cups_tea_first = 4

#### Looking at different ways to choose the cup of tea    

##### Using the math.comb(n, k) 

In this experiment the use of math.comb method returns the number of combinations of k unordered outcomes from n possibilities, without repetition. 

Checked https://www.w3schools.com/python/ref_math_comb.asp for clarification of how this method is applied.

The `math.comb()` function is called the *choose* function.

It is written and calculated as

$\binom{n}{k} = \frac{n!}{k!(n-k)!}$.

n = 8 (Number of cups of tea in total)

k = 4 (Number of cups of tea with milk first)

In [ ]:
# Number of ways of selecting four cups from eight.
ways = math.comb(no_cups, no_cups_milk_first)

# Show.
ways

70

Here we can see there are 70 possible combinations to choose the cups that have milk first. But this ones have not order.

So the next step is adding a label to the cups of tea like: `1`, `2`, `3`, `4`, `5`, `6`, `7`, and `8`.

Then we can select four cups to have milk in first.

For example, I might select `1`, `3`, `5`, and `7`.

How many ways are there to select four cups from eight?

Cups: `1 2 3 4 5 6 7 8`

Placeholders: `_ _ _ _`

In [ ]:
# Total number of ways of selecting four cups from eight, keeping them in order.
ways_ordered = 8 * 7 * 6 * 5

# Show.
ways_ordered

1680

In [ ]:
# Number of ways of shuffling four cups.
no_shuffles = 4 * 3 * 2 * 1

# Show.
no_shuffles

24

In [ ]:
# Number of combinations.
no_combs = ways_ordered // no_shuffles

# Show.
no_combs

70

So, right know we can see there are 1680 different selections ordered that the lady could do with 4 placeholder. How we can shuffle the 4 cups of tea in 24 different ways, this will leave as with 70 different ways of unique combinations unordered, that represents the ways the lady could pick the cups of tea with milk first. 

##### Using numpy in the experiment.


### Extending the experiment. 

In the extended version of the Lady Tasting Tea experiment, 12 cups are prepared: 8 with tea poured first and 4 with milk poured first. The participant asserts that they can correctly identify which cups were prepared by pouring tea first versus milk first

In [ ]:
# Number of cups of tea in total.
n_cups = 12

# Number of cups of tea with tea in first.
n_cups_tea_first = 8

# Number of cups of tea with milk in first.
n_cups_milk_first = 4


## Problem 2: Normal Distribution

## Problem 3: t-Tests

## Problem 4: ANOVA

## End

*************